library/src/main/java/me/zhanghai/android/patternlock/PatternUtils.java

```diff
@@ -39,10 +39,10 @@ private static String bytesToString(byte[] bytes) {
         return bytes;
     }
 
-    public static List<PatternView.Cell> bytesToPattern(byte[] bytes) {
+    public static List<PatternView.Cell> bytesToPattern(PatternView patternView, byte[] bytes) {
         List<PatternView.Cell> pattern = new ArrayList<>();
         for (byte b : bytes) {
-            pattern.add(PatternView.Cell.of(b / 3, b % 3));
```

Comment:
Please revert this file since we now have Cell.of again.

As a side note, to store and read patterns appropriately, we need the column count of pattern. We can have a method for that and overload it with one that takes a PatternView.